lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -381,6 +381,11 @@ class DocumentImpl extends NodeImpl {
   }
 
   _createElementWithCorrectElementInterface(name, namespace) {
+    let Constructor = null;
+
+    if (Constructor = this._global.window.customElements.get(name)) {
+      return idlUtils.tryImplForWrapper(new Constructor());
```

Comment:
@domenic 
Hi, I trying to test similar implementation by web-platform-tests, and I have strange problem.
If I will run in node code like 
```
import {JSDOM} from 'jsdom-wc';
const dom = new JSDOM(`<!DOCTYPE html>`);
global.HTMLElement = dom.window.HTMLElement;
global.window = dom.window;
global.document = dom.window.document;
class A extends HTMLElement {
    constructor() {
        super();
        console.log('AAAA');
    }
}
new A();
```
Everything works fine.
But if I want to run similar code from web-platform-tests I have a error: 
```
TypeError: Illegal constructor
    at CustomElement.HTMLElement (...\jsdom\living\generated\HTMLElement.js:14:10)
    at ...
    at Object.<anonymous> (http://w3c-test.org/custom-elements/attribute-changed-callback.html:6:31)
```
I putted similar code to create-jsdom.js just before `window.shimTest = () => {` and I see similar result.
The 'window.HTMLElement' is a function from ...\jsdom\living\generated\HTMLElement.js so calling of new is throwing same exception. 
In most of wpt there is no direct call of the elements constructors (instead of this document.creatElement is called) that why I think this is not occurs on other tests. But for custom-elements we have to call the constructor directly, and I do not know why on wpt this is no working. Any suggestion ?